dcpy/connectors/edm/recipes.py

```diff
@@ -35,26 +37,56 @@ class VersionStrategy(str, Enum):
     bump_latest_release = "bump_latest_release"
 
 
-class Dataset(BaseModel, use_enum_values=True):
+class DatasetType(str, Enum):
+    pg_dump = "pg_dump"
+    csv = "csv"
+
+
+_file_extensions_by_type = {"pg_dump": "sql", "csv": "csv"}
+
+
+class DataPreprocessor(BaseModel, use_enum_values=True, extra="forbid"):
+    module: str
+    function: str
+
+
+class Dataset(BaseModel, use_enum_values=True, extra="forbid"):
     name: str
     version: str | None = None
+    version_env_var: str | None = None
     import_as: str | None = None
+    file_type: DatasetType | None = None
+    preprocessor: DataPreprocessor | None = None
 
     def is_resolved(self):
         return self.version is not None and self.version != "latest"
 
+    @property
+    def file_name(self) -> str:
+        return f"{self.name}.{_file_extensions_by_type[self.file_type]}"
+
+    @property
+    def s3_key(self) -> str:
+        return f"datasets/{self.name}/{self.version}/{self.file_name}"
+
+
+class DatasetDefaults(BaseModel, use_enum_values=True):
```

Comment:
I understand how this is used in FacDB but I feel like in general this is complexity that I would prefer not to have in the build/recipe logic. Not 100% set on that, but it just feels like we're starting to nest more complex information in this class than is quite ideal, like maybe down the road, if this recipe is really the _start_ of a build, either the preprocessing has already happened, when we're ingesting data, or it's just part of the build. 

But again, not really relevant for this PR, just more stuff to think about as we iterate

Reply:
Since FacDB as is sort of relies on this structure.

Reply:
This is more about DataPreprocessor specifically - DatasetType obviously for now is quite useful